### PR TITLE
Add weekly report template

### PR DESCRIPTION
**Pull request by marekrozmus**

### CODE REVIEW

File: server/mailer/reports/weekly-content.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+const mailTemplate = require('./weekly-template');
+
+const styles = templateStyles({ maxWidth: 1024 });
+
+const tableRow = data => `<tr><td>${data}</td></tr>`;
+const tableHeaderCell = data => `<th style="${styles.thData}">${data}</th>`;
+const tableCell = data => `<td style="${styles.tdData}">${data}</td>`;
+
+const generateDataTable = ({ header, items, showAuthor, host }) => {
+  let content = '';
```
Comment (adamklepacz) [inline server/mailer/reports/weekly-content.js]: consider adding a new line after let and const declarations across this PR. We are doing this in the whole project
Regarding to this rule:
https://eslint.org/docs/rules/newline-after-var

File: server/mailer/reports/weekly-content.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  content += tableHeaderCell('Requested at');
+  content += '</tr>';
+  content += '</thead>';
+
+  content += '<tbody>';
+
+  for (let i = 0; i < items.length; i++) {
+    const { author, name, listName, listId, cohortName, requestedAt } = items[
+      i
+    ];
```
Comment (adamklepacz) [inline server/mailer/reports/weekly-content.js]: something wrong with formatting here??
 should be `items[i]` in one line

File: server/mailer/reports/weekly-content.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    `<h4 style="${styles.h4} margin-bottom: 0;">Your Todos</h4>`
+  );
+  content += generateDataTable({ items: todos, showAuthor: true, host });
+
+  return content;
+};
+
+const mailContent = ({ data, receiver, host, projectName }) => {
+  let content = '<p style="margin-bottom:0;">Here is your weekly report.</p>';
+
```
Comment (adamklepacz) [inline server/mailer/reports/weekly-content.js]: remove this line.
Group const and lets together

File: server/mailer/reports/weekly-content.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  if (todos) {
+    const result = generateTodosTable({ todos, host });
+    if (result) {
+      content += result;
+    }
+  }
+
+  if (requests) {
+    const result = generateRequestsContent({ requests, host });
```
Comment (adamklepacz) [inline server/mailer/reports/weekly-content.js]: consider adding a new line after cons declaration

File: server/mailer/reports/weekly-template.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
+const footer = require('../common/footer');
+const header = require('../common/header');
+
+const mailTemplate = ({ receiver, content, host, styles, projectName }) => {
```
Comment (adamklepacz) [inline server/mailer/reports/weekly-template.js]: unnecessary braces.

File: server/mailer/reports/weekly-content.js (hunk 2 above)
Comment (marekrozmus) [reply server/mailer/reports/weekly-content.js]: well, that's prettier

File: server/mailer/index.js [hunk 6]
```diff
@@ -1,6 +1,7 @@
 const SendGridMail = require('@sendgrid/mail');
 
 const mailTemplate = require('./mail-template');
+const weeklyReportContent = require('./reports/weekly-content');
```
[the remark this answers is not in the document]
Comment (marekrozmus) [reply server/mailer/index.js]: that is content generator which uses template inside - that is why such naming

**Pull request merged**